src/main/java/com/marek/onlinebookstore/service/cart/ShoppingCartServiceImpl.java

```diff
@@ -29,15 +29,15 @@ public class ShoppingCartServiceImpl implements ShoppingCartService {
     @Override
     public ShoppingCartDto findUserCart(User user) {
         return shoppingCartMapper.toDto(shoppingCartRepository
-                .findShoppingCartByUser(user));
+                .findShoppingCartByUser(user.getId()));
     }
 
     @Override
     public ShoppingCartDto addItem(User user, CartItemRequestDto cartItemRequestDto) {
         CartItem item = cartItemMapper.toModel(cartItemRequestDto);
         item.setBook(bookRepository.findById(item.getBook().getId())
```

Comment:
The book's existence in the database is assumed. It would be good to add error handling for the case when the book is not found in the database.